src/inference/src/ie_core.cpp

```diff
@@ -699,10 +699,25 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             opNames.emplace(op->get_friendly_name());
 
         for (const auto& op : func->get_ops()) {
-            if (opNames.find(op->get_friendly_name()) == opNames.end() ||
-                (!res.supportedLayersMap.count(op->get_friendly_name()) &&
-                 std::dynamic_pointer_cast<ngraph::op::Constant>(op)))
+            if (opNames.find(op->get_friendly_name()) == opNames.end()) {
                 res.supportedLayersMap[op->get_friendly_name()] = defDevice;
+            }
+        }
+
+        for (const auto& op : func->get_ops()) {
```

Comment:
@amolotko is it important that order of ops should be specific? E.g. if operation A has input B, but A comes first in this list, but B is supported as well - we can have situation when A is not marked as supported

Reply:
The situation you described will be processed correctly. Perhaps you mean case when operation A has user B which should be marked as supported in future but now no and node A comes first. This situation is processed in loop above where all needed nodes are marked as supported. And as far I know constant nodes are not users of constant nodes and situation was described by me above can not be.